builder/ghc-for-component-wrapper.nix

```diff
@@ -0,0 +1,90 @@
+# The ghcForComponent function wraps ghc so that it is configured with
+# the package database of all dependencies of a given component.
+# It has been adapted from the ghcWithPackages wrapper in nixpkgs.
+#
+# This wrapper exists so that nix-shells for components will have a
+# GHC automatically configured with the dependencies in the package
+# database.
+
+{ lib, stdenv, ghc, runCommand, lndir, makeWrapper
+}:
+
+{ componentName  # Full derivation name of the component
+, configFiles    # The component's "config" derivation
+}:
+
+let
+  isGhcjs        = ghc.isGhcjs or false;
+  ghcCommand'    = if isGhcjs then "ghcjs" else "ghc";
+  ghcCommand     = "${ghc.targetPrefix}${ghcCommand'}";
+  ghcCommandCaps = lib.toUpper ghcCommand';
+  libDir         = "$out/lib/${ghcCommand}-${ghc.version}";
+  docDir         = "$out/share/doc/ghc/html";
+  packageCfgDir  = "${libDir}/package.conf.d";
+
+in runCommand "${componentName}-${ghc.name}" {
+  preferLocalBuild = true;
+  passthru = {
+    inherit (ghc) version meta;
+    baseGhc = ghc;
+  };
+} (
+  ''
+    . ${makeWrapper}/nix-support/setup-hook
+
+    # Start with a ghc...
+    mkdir -p $out/bin
+    ${lndir}/bin/lndir -silent ${ghc} $out
+    # ... remove all of the package directories
+    rm -rf ${libDir}/*/
+    # ... but retain the lib/ghc/bin directory. This contains `unlit' and friends.
+    ln -s ${ghc}/lib/${ghcCommand}-${ghc.version}/bin ${libDir}
```

Comment:
It's hard to see because I've renamed the file but @angerman please note that I have changed the unlit&friends fix to link the `bin` directory rather than using `lndir`.

Reply:
@rvl should be identical. So I'm ok wither way :-)